pkg/descheduler/strategies/lownodeutilization.go

```diff
@@ -215,30 +220,96 @@ func getNodeUsage(
 			nodeCapacity = node.Status.Allocatable
 		}
 
-		nodeUsageList = append(nodeUsageList, NodeUsage{
-			node:    node,
-			usage:   nodeUtilization(node, pods),
-			allPods: pods,
-			// A threshold is in percentages but in <0;100> interval.
-			// Performing `threshold * 0.01` will convert <0;100> interval into <0;1>.
-			// Multiplying it with capacity will give fraction of the capacity corresponding to the given high/low resource threshold in Quantity units.
-			lowResourceThreshold: map[v1.ResourceName]*resource.Quantity{
-				v1.ResourceCPU:    resource.NewMilliQuantity(int64(float64(lowThreshold[v1.ResourceCPU])*float64(nodeCapacity.Cpu().MilliValue())*0.01), resource.DecimalSI),
-				v1.ResourceMemory: resource.NewQuantity(int64(float64(lowThreshold[v1.ResourceMemory])*float64(nodeCapacity.Memory().Value())*0.01), resource.BinarySI),
-				v1.ResourcePods:   resource.NewQuantity(int64(float64(lowThreshold[v1.ResourcePods])*float64(nodeCapacity.Pods().Value())*0.01), resource.DecimalSI),
-			},
-			highResourceThreshold: map[v1.ResourceName]*resource.Quantity{
-				v1.ResourceCPU:    resource.NewMilliQuantity(int64(float64(highThreshold[v1.ResourceCPU])*float64(nodeCapacity.Cpu().MilliValue())*0.01), resource.DecimalSI),
-				v1.ResourceMemory: resource.NewQuantity(int64(float64(highThreshold[v1.ResourceMemory])*float64(nodeCapacity.Memory().Value())*0.01), resource.BinarySI),
-				v1.ResourcePods:   resource.NewQuantity(int64(float64(highThreshold[v1.ResourcePods])*float64(nodeCapacity.Pods().Value())*0.01), resource.DecimalSI),
-			},
+		usage := nodeUtilization(node, pods)
+
+		// A threshold is in percentages but in <0;100> interval.
+		// Performing `threshold * 0.01` will convert <0;100> interval into <0;1>.
+		// Multiplying it with capacity will give fraction of the capacity corresponding to the given high/low resource threshold in Quantity units.
+		lowResourceThreshold := make(map[v1.ResourceName]*resource.Quantity)
+		highResourceThreshold := make(map[v1.ResourceName]*resource.Quantity)
+
+		if strategyConfig.UseDeviationThresholds {
+			// we'll have to compute the absolute thresholds later because
+			// we need to compute the averages over all nodes
+			totalCPUUsagePercent += float64(usage[v1.ResourceCPU].MilliValue()) / float64(nodeCapacity.Cpu().MilliValue()) * 100.0
+			totalMemoryUsagePercent += float64(usage[v1.ResourceMemory].Value()) / float64(nodeCapacity.Memory().Value()) * 100.0
+			totalPodsUsagePercent += float64(usage[v1.ResourcePods].Value()) / float64(nodeCapacity.Pods().Value()) * 100.0
+		} else {
+			lowResourceThreshold = map[v1.ResourceName]*resource.Quantity{
+				v1.ResourceCPU:    resource.NewMilliQuantity(int64(float64(strategyConfig.Thresholds[v1.ResourceCPU])*float64(nodeCapacity.Cpu().MilliValue())*0.01), resource.DecimalSI),
+				v1.ResourceMemory: resource.NewQuantity(int64(float64(strategyConfig.Thresholds[v1.ResourceMemory])*float64(nodeCapacity.Memory().Value())*0.01), resource.BinarySI),
```

Comment:
Is `resource.BinarySI` correct here? I took that from the existing code, but I'm surprised that this is different from `resource.DecimalSI` for the CPU and pods. Not sure about the purpose/meaning of that parameter though.

Reply:
Yes, they're different because for CPU, 5 cores = 5 * 1000m, but for Memory, 5GiB = 5 * 1024 MiB.